### PR TITLE
Dev.del/feat studio e2e

### PR DESCRIPTION
**Pull request by deltork**
### PR Goal? 
Bring end to end tests to Studio-Web

### Feedback sought? 
sanity

### Priority? 
High

### Tests added? 
`packages\studio-web\tests\studio-web.spec`

### How to test? 
check out the branch and run `npx nx e2e studio-web`

### Confidence? 
High

### Version change? 
n/a

The testing the editor will be another PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,6 +68,7 @@
     "jszip": "^3.10.1",
     "mime": "^4.0.1",
     "ngx-toastr": "^18.0.0",
+    "root": "file:",
```
Comment (joanise) [inline package.json:67]: This is weird, what does this mean?

File: packages/studio-web/.github/workflows/playwright.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+name: Playwright Tests
```
Comment (joanise) [inline packages/studio-web/.github/workflows/playwright.yml]: Why is this file here? workflows needs to be from the root of the repo in the `.github` directory, there, no?

File: .github/workflows/end-to-end-tests.yml [hunk 3]
```diff
@@ -56,3 +69,81 @@ jobs:
...
+          # wait for the API to be up
+          curl --retry 20 --retry-delay 1 --retry-all-errors http://localhost:8000/api/v1/langs
+      - name: Install everything
+        run: npm install
+      - name: Install dependencies
+        run: npm ci
+      - name: Run studio-web in the background
+        run: |
+          npx nx build web-component
+          npx nx run-many --targets=serve,serve-fr,serve-es --projects=web-component,studio-web --parallel 6 &
```
Comment (joanise) [inline .github/workflows/end-to-end-tests.yml:86]: Unless we actually exercise fr and es, I would drop serve-fr and serve-es from here.

**Review by joanise — COMMENTED** [review]
I'm going to want to test this locally a bit, but just some comments for now.

File: package.json (hunk 1 above)
Comment (deltork) [reply package.json:67]: argh! something in my configuration keeps adding them! I will remove it

File: .github/workflows/end-to-end-tests.yml (hunk 3 above)
Comment (deltork) [reply .github/workflows/end-to-end-tests.yml:86]: We do see `packages/studio-web/tests/studio-web/check-page-1.spec.ts`

File: packages/studio-web/playwright.config.ts [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ */
+// import dotenv from 'dotenv';
+// import path from 'path';
+// dotenv.config({ path: path.resolve(__dirname, '.env') });
+
+/**
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig({
+  timeout: 2 * 60 * 1000,
```
Comment (joanise) [inline packages/studio-web/playwright.config.ts]: Make this about 15s or no more than 30s.

File: package.json [hunk 5]
```diff
@@ -40,10 +42,14 @@
...
     "nx": "18.3.4",
+    "nyc": "^17.1.0",
     "prettier": "^3.2.5",
     "pretty-quick": "^4.0.0",
+    "source-map-support": "^0.5.21",
     "ts-jest": "^29",
-    "tsx": "^4.7.3"
+    "ts-node": "^10.9.2",
+    "tsx": "^4.7.3",
+    "v8-to-istanbul": "^9.3.0"
```
Comment (joanise) [inline package.json]: remove this for now

File: package.json [hunk 6]
```diff
@@ -9,12 +9,14 @@
     "@angular/localize": "^17",
     "@commitlint/cli": "^19.2.1",
     "@commitlint/config-conventional": "^19.1.0",
+    "@istanbuljs/nyc-config-typescript": "^1.0.2",
```
Comment (joanise) [inline package.json]: remove too

File: package.json [hunk 7]
```diff
@@ -40,10 +42,14 @@
     "karma-jasmine-html-reporter": "^2.1.0",
     "ng-packagr": "^17",
     "nx": "18.3.4",
+    "nyc": "^17.1.0",
```
Comment (joanise) [inline package.json]: remove

File: package.json [hunk 8]
```diff
@@ -40,10 +42,14 @@
...
     "prettier": "^3.2.5",
     "pretty-quick": "^4.0.0",
+    "source-map-support": "^0.5.21",
```
Comment (joanise) [inline package.json]: ?

File: packages/studio-web/.gitignore [hunk 9]
```diff
@@ -35,3 +35,4 @@ e2e/*.map
 
 # End of https://www.toptal.com/developers/gitignore/api/angular
 *~
+.nyc_output
```
Comment (joanise) [inline packages/studio-web/.gitignore]: undo too?

File: packages/studio-web/.nycrc [hunk 10]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (joanise) [inline packages/studio-web/.nycrc]: remove this file, re-add it if and when we add cov

File: packages/studio-web/package.json [hunk 11]
```diff
@@ -9,7 +9,10 @@
     "helpme": "echo This project is part of a monorepo managed using nx. Run the targets in project.json using npx nx target studio-web at the root of the monorepo.",
     "ng": "ng",
     "test:ng": "ng test",
-    "test:once": "ng test --watch=false --browsers ChromeHeadlessCI"
+    "test:once": "ng test --watch=false --browsers ChromeHeadlessCI",
+    "e2e": "playwright test",
+    "nyc": "nyc --reporter=html --reporter=text-summary playwright test ",
```
Comment (joanise) [inline packages/studio-web/package.json]: ditto

File: packages/studio-web/playwright.config.ts [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+
+    /* Test against branded browsers. */
+        // {
+        //   name: 'Microsoft Edge',
+        //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
+        // },
+        // {
+        //   name: 'Google Chrome',
+        //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
+        // },
```
Comment (joanise) [inline packages/studio-web/playwright.config.ts]: make indentation consistent with the enclosing block

File: package.json (hunk 1 above)
Comment (joanise) [reply package.json:67]: This slipped back in.

**Review by joanise — APPROVED** [review]
This is great, and I think we can merge it already.
Trivial detail: the "root": "file" line is back in. Is it innocuous? Maybe we could just leave it in if  it doesn't hurt, I just wish I understood what it means. There's another "file" entry, too, for studio-web depending on web-component: file. Again, what does this mean?

Locally, three tests don't pass on Firefox, mostly due to exact alignments, but I don't think we need to hold the PR back for this.

File: package.json (hunk 1 above)
Comment (deltork) [reply package.json:67]: I disabled all my plugins, but they keep popping up!

**Comment by deltork** [comment]
> This is great, and I think we can merge it already. Trivial detail: the "root": "file" line is back in. Is it innocuous? Maybe we could just leave it in if it doesn't hurt, I just wish I understood what it means. There's another "file" entry, too, for studio-web depending on web-component: file. Again, what does this mean?
> 
It's something my VS Code keeps adding
> Locally, three tests don't pass on Firefox, mostly due to exact alignments, but I don't think we need to hold the PR back for this.
I slightly modified the evaluation of the output to help with that issue
